overseer/src/lib.rs

```diff
@@ -471,6 +489,7 @@ where
 	/// # }); }
 	/// ```
 	pub fn new(
+		leaves: &[Hash],
```

Comment:
```suggestion
		leaves: impl Into<Vec<Hash>>,
```

You take a slice and convert it into a `Vec` anyway. This makes the user either pass a `Vec` directly or still pass the slice.